packages/hardhat-core/src/internal/core/providers/accounts.ts

```diff
@@ -249,54 +246,118 @@ export class LocalAccountsProvider extends ProviderWrapperWithChainId {
-      "@nomicfoundation/ethereumjs-tx"
+    const { bytesToHex, bytesToInt, bytesToBigInt } = await import(
+      "@ethereumjs/util"
     );
-
-    const { Common } = await import("@nomicfoundation/ethereumjs-common");
+    const { addr, Transaction } = await import("micro-eth-signer");
 
     const txData = {
       ...transactionRequest,
       gasLimit: transactionRequest.gas,
     };
 
-    // We don't specify a hardfork here because the default hardfork should
-    // support all possible types of transactions.
-    // If the network doesn't support a given transaction type, then the
-    // transaction it will be rejected somewhere else.
-    const common = Common.custom({ chainId, networkId: chainId });
+    const accessList = txData.accessList?.map(({ address, storageKeys }) => {
+      return {
+        address: addr.addChecksum(bytesToHex(address)),
+        storageKeys:
+          storageKeys !== null ? storageKeys.map((k) => bytesToHex(k)) : [],
+      };
+    });
+
+    const authorizationList = txData.authorizationList?.map(
+      ({ chainId: authChainId, address, nonce, yParity, r, s }) => {
+        return {
+          chainId: authChainId,
+          address: addr.addChecksum(bytesToHex(address)),
+          nonce,
+          yParity: bytesToInt(yParity),
+          r: bytesToBigInt(r),
+          s: bytesToBigInt(s),
+        };
+      }
+    );
 
-    // we convert the access list to the type
-    // that AccessListEIP2930Transaction expects
-    const accessList = txData.accessList?.map(
-      ({ address, storageKeys }) => [address, storageKeys] as [Buffer, Buffer[]]
+    let checksummedAddress;
+    if (txData.to === undefined || txData.to === null) {
+      // This scenario arises during contract deployment. The npm package "micro-eth-signer" does not support
+      // null or undefined addresses. Therefore, these values must be converted to "0x", the expected format.
+      checksummedAddress = "0x";
+    } else {
+      checksummedAddress = addr.addChecksum(
+        bytesToHex(txData.to).toLowerCase()
+      );
+    }
+
+    assertHardhatInvariant(
+      txData.nonce !== undefined,
+      "nonce should be defined"
     );
 
     let transaction;
-    if (txData.maxFeePerGas !== undefined) {
-      transaction = FeeMarketEIP1559Transaction.fromTxData(
-        {
-          ...txData,
-          accessList,
-          gasPrice: undefined,
-        },
-        { common }
+    if (authorizationList !== undefined) {
+      console.log("eip7702");
+      assertHardhatInvariant(
+        txData.maxFeePerGas !== undefined,
+        "maxFeePerGas should be defined"
       );
+
+      transaction = Transaction.prepare({
+        type: "eip7702",
+        to: checksummedAddress,
+        nonce: txData.nonce,
+        chainId: txData.chainId ?? normalizeToBigInt(chainId),
+        value: txData.value !== undefined ? txData.value : 0n,
+        data: txData.data !== undefined ? bytesToHex(txData.data) : "",
```

Comment:
Same for some fields below.

```suggestion
        value: txData.value ?? 0n,
        data: txData.data ?? "",
```

Reply:
I also find it better, unless `null` has some special meaning here.